### PR TITLE
SITL parameters documentation update

### PR DESCRIPTION
**Pull request by YuxinPan**
Partially addressing #22903 

Passed Sim helper check and CI check:
image

image

Will squash commits once approved.

@peterbarker

### CODE REVIEW

File: libraries/SITL/SITL.cpp [hunk 1]
```diff
@@ -65,8 +65,14 @@ SIM *SIM::_singleton = nullptr;
...
 const AP_Param::GroupInfo SIM::var_info[] = {
-    
+
+    // @Param: DRIFT_SPEED
+    // @DisplayName: Gyro drift speed
+    // @Description: Gyro drift rate of change in degrees/second/minute
     AP_GROUPINFO("DRIFT_SPEED",    5, SIM,  drift_speed, 0.05f),
+    // @Param: DRIFT_TIME
+    // @DisplayName: Gyro drift time
+    // @Description: Gyro drift duration of one full drift cycle (period in minutes)
```
Comment (peterbarker) [inline libraries/SITL/SITL.cpp:75]: Is this really in minutes?

File: libraries/SITL/SITL.cpp [hunk 2]
```diff
@@ -65,8 +65,14 @@ SIM *SIM::_singleton = nullptr;
 
 // table of user settable parameters
 const AP_Param::GroupInfo SIM::var_info[] = {
-    
+
+    // @Param: DRIFT_SPEED
+    // @DisplayName: Gyro drift speed
+    // @Description: Gyro drift rate of change in degrees/second/minute
```
Comment (peterbarker) [inline libraries/SITL/SITL.cpp:71]: Are these *really* in minutes?!  That's rather odd.  Not saying you're wrong, but please make sure its right!

Also, we can put units into separate fields, like this:

```suggestion
    // @Description: Gyro drift rate of change
    // @Units: deg/s/s
```

Known units are here: https://github.com/ardupilot/ardupilot/blob/master/Tools/autotest/param_metadata/param.py#L66

File: libraries/SITL/SITL.cpp [hunk 3]
```diff
@@ -111,6 +117,10 @@ const AP_Param::GroupInfo SIM::var_info[] = {
     // @Units: V
     // @User: Advanced
     AP_GROUPINFO("BATT_VOLTAGE",  19, SIM,  batt_voltage,  12.6f),
+    // @Param: BATT_CAP_AH
+    // @DisplayName: Simulated battery capacity
+    // @Description: Simulated battery capacity in Amp hours
```
Comment (peterbarker) [inline libraries/SITL/SITL.cpp:122]:
```suggestion
    // @Description: Simulated battery capacity
    // @Units: Ah
```

Will need to add a unit to the above table of known units

File: libraries/SITL/SITL.cpp [hunk 4]
```diff
@@ -168,10 +178,25 @@ const AP_Param::GroupInfo SIM::var_info[] = {
     // @Description: Opflow data delay
     // @Units: ms
     AP_GROUPINFO("FLOW_DELAY",    36, SIM,  flow_delay, 0),
+    // @Param: ADSB_COUNT
+    // @DisplayName: Number of ADSB aircrafts
+    // @Description: Total number of ADSB simulated aircrafts
```
Comment (peterbarker) [inline libraries/SITL/SITL.cpp:183]:
```suggestion
    // @Description: Total number of ADSB simulated aircraft
```

plural of aircraft is... aircraft

File: libraries/SITL/SITL.cpp [hunk 5]
```diff
@@ -168,10 +178,25 @@ const AP_Param::GroupInfo SIM::var_info[] = {
...
     AP_GROUPINFO("ADSB_COUNT",    45, SIM,  adsb_plane_count, -1),
+    // @Param: ADSB_RADIUS
+    // @DisplayName: ADSB radius stddev of another aircraft
```
Comment (peterbarker) [inline libraries/SITL/SITL.cpp:186]: Is this right?  I thought this was just a maximum?!

File: libraries/SITL/SITL.cpp [hunk 6]
```diff
@@ -168,10 +178,25 @@ const AP_Param::GroupInfo SIM::var_info[] = {
...
+    // @Units: m
     AP_GROUPINFO("ADSB_RADIUS",   46, SIM,  adsb_radius_m, 10000),
+    // @Param: ADSB_ALT
+    // @DisplayName: ADSB altitude of another aircraft
+    // @Description: Simulated ADSB altitude of another aircraft
+    // @Units: m
     AP_GROUPINFO("ADSB_ALT",      47, SIM,  adsb_altitude_m, 1000),
     AP_GROUPINFO("PIN_MASK",      50, SIM,  pin_mask, 0),
+    // @Param: ADSB_TX
+    // @DisplayName: ADSB transceiever enable
```
Comment (peterbarker) [inline libraries/SITL/SITL.cpp:197]:
```suggestion
    // @DisplayName: ADSB transmit enable
```

... it's the transmit part of the transceiver which is important here

File: Tools/autotest/vehicle_test_suite.py [hunk 7]
```diff
@@ -2559,64 +2559,51 @@ def test_adsb_send_threatening_adsb_message(self, here, offset_ne=None):
...
+            "SIM_ACC_TRIM_Z",    # not listed in SITL.cpp
+            "SIM_ARSPD2_OFS",    # not listed in SITL.cpp
+            "SIM_ARSPD2_RND",    # not referenced anywhere
+            "SIM_ARSPD_OFS",     # not referenced anywhere
+            "SIM_ARSPD_RND",     # not listed in SITL.cpp
+            "SIM_BAR2_DELAY",    # not referenced anywhere
+            "SIM_BAR2_DISABLE",  # not listed in SITL.cpp
+            "SIM_BAR2_DRIFT",    # not referenced anywhere
+            "SIM_BAR2_FREEZE",   # not listed in SITL.cpp
+            "SIM_BAR2_WCF_BAK",  # not referenced anywhere
```
Comment (peterbarker) [inline Tools/autotest/vehicle_test_suite.py:2573]: Everything in this list is used - we have a sanity check for that ;-)

In this case the parameters are coming from in here: https://github.com/ardupilot/ardupilot/blob/pr%2Fstop-using-python-is-python3/libraries/SITL/SITL.cpp#L589

Fixing this requires moving the BaroParm structure to a new file, probably SITL_Baro.cpp .  @antholuo did the same thing to document airspeed by creating SITL_Airspeed.cpp, so use that as a model

File: Tools/autotest/vehicle_test_suite.py [hunk 8]
```diff
@@ -2559,64 +2559,51 @@ def test_adsb_send_threatening_adsb_message(self, here, offset_ne=None):
...
+            "SIM_BARO_DISABLE",  # not listed in SITL.cpp
+            "SIM_BARO_FREEZE",   # not referenced anywhere
+            "SIM_BARO_WCF_BAK",  # not listed in SITL.cpp
+            "SIM_BARO_WCF_DN",   # not listed in SITL.cpp
+            "SIM_BARO_WCF_FWD",  # not listed in SITL.cpp
+            "SIM_BARO_WCF_LFT",  # not listed in SITL.cpp
+            "SIM_BARO_WCF_RGT",  # not listed in SITL.cpp
+            "SIM_BARO_WCF_UP",   # not listed in SITL.cpp
+            "SIM_FTOWESC_ENA",   # not listed in SITL.cpp
+            "SIM_FTOWESC_POW",   # not listed in SITL.cpp
```
Comment (peterbarker) [inline Tools/autotest/vehicle_test_suite.py:2599]: These will be in SIM_FETTechOneWireESC

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: libraries/SITL/SITL.cpp (hunk 3 above)
Comment (YuxinPan) [reply libraries/SITL/SITL.cpp:122]: `Ah` added  as known unit

File: libraries/SITL/SITL.cpp (hunk 2 above)
Comment (YuxinPan) [reply libraries/SITL/SITL.cpp:71]: It is indeed uncommon. I referred to the `SITL.h` where the comment states:     `AP_Float drift_speed; // degrees/second/minute`
Also, I see `    double minutes = fmod(AP_HAL::micros64() / 60.0e6, period);
    if (minutes < period/2) {
        return minutes * ToRad(sitl->drift_speed);
    }
    return (period - minutes) * ToRad(sitl->drift_speed);`

File: libraries/SITL/SITL.cpp (hunk 1 above)
Comment (YuxinPan) [reply libraries/SITL/SITL.cpp:75]: Appears to be: `AP_Float drift_time;  // period in minutes`

File: libraries/SITL/SITL.cpp (hunk 5 above)
Comment (YuxinPan) [reply libraries/SITL/SITL.cpp:187]: I can see the usage: `Aircraft::rand_normal(0, _sitl->adsb_radius_m);`
where `Aircraft::rand_normal` is defined as `double Aircraft::rand_normal(double mean, double stddev)`

File: Tools/autotest/vehicle_test_suite.py (hunk 8 above)
Comment (YuxinPan) [reply Tools/autotest/vehicle_test_suite.py]: I see, I have removed those param comments

File: Tools/autotest/vehicle_test_suite.py (hunk 7 above)
Comment (YuxinPan) [reply Tools/autotest/vehicle_test_suite.py]: Thanks I can follow the approach in #25005. Do you want that to be a separate PR or included in this one?

File: Tools/autotest/param_metadata/param.py [hunk 9]
```diff
@@ -117,13 +117,14 @@ def has_param(self, pname):
              'dB'      : 'decibel'               ,
 # compound
 
-             'kB'      : 'kilobytes'                ,
```
Comment (peterbarker) [inline Tools/autotest/param_metadata/param.py:120]: Bad whitespace change

File: Tools/autotest/param_metadata/param.py [hunk 10]
```diff
@@ -117,13 +117,14 @@ def has_param(self, pname):
...
+             'kB'      : 'kilobytes'               ,
              'MB'      : 'megabyte'                ,
              'm.m/s/s' : 'square meter per square second',
              'deg/m/s' : 'degrees per meter per second'  ,
              'm/s/m'   : 'meters per second per meter'   , # Why not use Hz here ????
              'mGauss/A': 'milligauss per ampere' ,
-             'mAh'    : 'milliampere hour'      ,
+             'mAh'     : 'milliampere hour'      ,
```
Comment (peterbarker) [inline Tools/autotest/param_metadata/param.py:126]: Bad whitespace change

File: libraries/SITL/SITL.cpp (hunk 2 above)
Comment (peterbarker) [reply libraries/SITL/SITL.cpp:71]: Great, thanks!

Sometimes the hard-to-believe things just make you do more work :-)

**Review by peterbarker — APPROVED** [review]
LGTM

File: Tools/autotest/vehicle_test_suite.py (hunk 7 above)
Comment (peterbarker) [reply Tools/autotest/vehicle_test_suite.py]: Separate PR makes sense.  Makes these easy to review!

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**